src/ert/_c_wrappers/enkf/site_config.py

```diff
@@ -1,59 +1,119 @@
 import os
+from dataclasses import dataclass
 
-from cwrap import BaseCClass
-
-from ert._c_wrappers import ResPrototype
+from ert._c_wrappers.config import ConfigContent
 from ert._c_wrappers.enkf.config_keys import ConfigKeys
-from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob
-
-
-class SiteConfig(BaseCClass):
-    TYPE_NAME = "site_config"
-    _alloc = ResPrototype("void* site_config_alloc(config_content)", bind=False)
-    _site_config_alloc_default = ResPrototype(
-        "void* site_config_alloc_default()", bind=False
-    )
-    _alloc_full = ResPrototype(
-        "void* site_config_alloc_full(ext_joblist, env_varlist)", bind=False
-    )
-    _free = ResPrototype("void site_config_free( site_config )")
-    _get_installed_jobs = ResPrototype(
-        "ext_joblist_ref site_config_get_installed_jobs(site_config)"
-    )
-    _get_license_root_path = ResPrototype(
-        "char* site_config_get_license_root_path(site_config)"
-    )
-    _get_location = ResPrototype("char* site_config_get_location()", bind=False)
-    _get_config_file = ResPrototype("char* site_config_get_config_file(site_config)")
-    _get_env_var_list = ResPrototype(
-        "env_varlist_ref site_config_get_env_varlist(site_config)"
-    )
-
-    def __init__(self, config_content=None):
-
-        if config_content:
-            c_ptr = self._alloc(config_content)
-        else:
-            c_ptr = self._site_config_alloc_default()
-
-        if c_ptr is None:
-            raise ValueError("Failed to construct SiteConfig instance.")
-
-        super().__init__(c_ptr)
+from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob, ExtJoblist
+
+
+@dataclass
+class SiteConfig:
+    job_list: ExtJoblist
+    var_list: EnvironmentVarlist
+
+    @classmethod
+    def _add_config_content(
+        cls,
+        license_root_path: str,
+        site_config_content: ConfigContent,
```

Comment:
should this one be called `site_config_content`, or perhaps just `config_content`? i feel it shadows the first of the pair `site_config_content` and `user_config_content` (which is actually just called `config_content` in `from_config_content`) - maybe you can change that one to `user_config_content` as well? gonna add a comment down there as well

Reply:
I was a bit hyper-caffinated yesterday and suggested that I would address a refactor of this today. I might have been curt, and if so I am sorry.

My general idea was that we could separate the initialization into the one list and two dictionaries we need:

```python
    @staticmethod
    def _jobs_from_config_content(
        config_content: ConfigContent, license_root_path: str
    ):
        jobs = {}
        if config_content.hasKey(ConfigKeys.INSTALL_JOB):
            for args in iter(config_content[ConfigKeys.INSTALL_JOB]):
                if not os.path.isfile(args[1]):
                    logger.warn("Unable to locate job file {}", args)
                    continue
                try:
                    jobs[args[0]] = ExtJob(
                        config_file=args[1],
                        private=False,
                        name=args[0],
                        license_root_path=license_root_path,
                    )
                except (ValueError, OSError):
                    logger.warn("Unable to create job from {}", args)

        if config_content.hasKey(ConfigKeys.INSTALL_JOB_DIRECTORY):
            for args in iter(config_content[ConfigKeys.INSTALL_JOB_DIRECTORY]):
                job_path = args[0]
                if not os.path.isdir(job_path):
                    print(f"WARNING: Unable to locate job directory {job_path}")
                    continue
                files = os.listdir(job_path)
                for file_name in files:
                    full_path = os.path.join(job_path, file_name)
                    if os.path.isfile(full_path):
                        try:
                            new_job = ExtJob(
                                config_file=full_path,
                                private=False,
                                license_root_path=license_root_path,
                            )
                            jobs[new_job.name] = new_job
                        except (ValueError, OSError):
                            logger.warn(
                                ": Unable to create job from {} " "in directory {}",
                                full_path,
                                job_path,
                            )
        return jobs

    @staticmethod
    def _setenvs_from_config_content(config_content: ConfigContent):
        return {elem[0]: elem[1] for elem in config_content[ConfigKeys.SETENV]}

    @staticmethod
    def _updatepaths_from_config_content(config_content: ConfigContent):
        return {elem[0]: elem[1] for elem in config_content[ConfigKeys.UPDATE_PATH]}

    @staticmethod
    def _license_root_path(
        site_config_content: ConfigContent, user_config_content: ConfigContent
    ) -> str:
        license_root_path = None
        if site_config_content.hasKey(ConfigKeys.LICENSE_PATH):
            license_root_path = site_config_content.getValue(ConfigKeys.LICENSE_PATH)
        if user_config_content.hasKey(ConfigKeys.LICENSE_PATH):
            user_license_root_path = user_config_content.getValue(
                ConfigKeys.LICENSE_PATH
            )
            license_root_path_site = os.path.realpath(user_license_root_path)
            license_root_path = os.path.join(
                license_root_path_site, os.getenv("USER"), str(os.getpid())
            )
        return license_root_path

    @classmethod
    def from_config_content(
        cls, config_content: ConfigContent, site_config_content: ConfigContent
    ):
        license_root_path = cls._license_root_path(site_config_content, config_content)
        return cls(
            ExtJoblist(
                cls._jobs_from_config_content(config_content, license_root_path).update(
                    cls._jobs_from_config_content(
                        site_config_content, license_root_path
                    )
                )
            ),
            EnvironmentVarlist(
                vars=cls._setenvs_from_config_content(config_content).update(
                    cls._setenvs_from_config_content(site_config_content)
                ),
                paths=cls._updatepaths_from_config_content(config_content).update(
                    cls._updatepaths_from_config_content(site_config_content)
                ),
            ),
        )

```
That way I think we will have a shot at making from_dict and from_content take more or less the same path.

Reply:
I think we can leave this for later since we want to refactor this further.